f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -212,6 +218,72 @@ def create(self, loadbalancer, service, **kwargs):
             service["loadbalancer"], service)
         self._post_create(service)
 
+    def _create(self, bigip, payload, loadbalancer, service, **kwargs):
+
+        # create irule
+        if 'bandwidth' in loadbalancer.keys() and \
+           loadbalancer['bandwidth']:
+            self.__add_bwc(bigip, loadbalancer, service, False)
+
+        super(LoadBalancerManager, self)._create(bigip, payload,
+                                                 loadbalancer, service)
+
+    def _delete(self, bigip, payload, loadbalancer, service, **kwargs):
+
+        # create irule
+        if 'bandwidth' in loadbalancer.keys() and \
+           loadbalancer['bandwidth']:
+            LOG.debug("bandwidth exists")
+            # return
+            profile = self._create_bwc_irule_profile(loadbalancer, True)
+            super(LoadBalancerManager, self).\
+                _delete(bigip, profile, None, None,
+                        type="irule-profile",
+                        helper=self.irule_helper)
+
+            profile = self._create_bwc_profile(loadbalancer, 0)
+            super(LoadBalancerManager, self).\
+                _delete(bigip, profile, None, None, type="bwc-profile",
+                        helper=self.bwc_profile_helper)
+
+        super(LoadBalancerManager, self)._delete(bigip, payload,
+                                                 loadbalancer, service)
+
+    def _create_bwc_profile(self, loadbalancer, bandwidth):
+        # create irule profile
+        bwc_profile = {}
+        bwc_profile['partition'] = self.driver.service_adapter.\
+            get_folder_name(loadbalancer['tenant_id'])
+        bwc_profile['name'] = 'bwc_' + 'profile_' + loadbalancer['id']
+        if bandwidth > 0:
+            bwc_profile['maxRate'] = bandwidth
+        return bwc_profile
+
+    def __get_bwc_profile_name(self, loadbalancer):
+
+        profile_name = '/' + self.driver.service_adapter.\
+            get_folder_name(loadbalancer['tenant_id']) + \
+            '/' + 'bwc_' + 'profile_' + loadbalancer['id']
+        return profile_name
+
+    def __get_irule_profile_name(self, loadbalancer):
```

Comment:
This terminology may looks confused. iRule, profile and bwc policy are different things, and you are create an iRule whose content has a reference to a bwc policy. Maybe we can name it a little better. Such kind of confusing name also exists in some other places.

Reply:
accept.